portafly/portafly_api.yml

```diff
@@ -0,0 +1,127 @@
+
+    get:
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Application'
+              examples:
+                Application List Example:
+                  value:
+                  - id: "1"
+                    created_at: 2020-01-10T09:30Z
+                    updated_at: 2020-01-10T09:30Z
+                    state: live
+                    user_account_id: "3"
+                    service_id: "2"
+                    name: Default App
+                    description: Default Application
+                  - id: "2"
+                    created_at: 2020-02-10T09:30Z
+                    updated_at: 2020-02-10T09:30Z
+                    state: live
+                    user_account_id: "3"
+                    service_id: "2"
+                    name: Another App
+                    description: User created example application
+          description: Successful response - returns an array of `Application`.
+      summary: Get a list of Applications
+      description: |+
+        Returns the list of applications across all services. Note that applications are scoped by service and can be paginated.
+
+  /applications/{applicationId}:
+    summary: Path used to manage a single Application.
+    description: The REST endpoint/path used to get, update, and delete single instances
+      of an `Application`.  This path contains `GET`, `PUT`, and `DELETE` operations
+      used to perform the get, update, and delete tasks, respectively.
+    get:
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Application'
+          description: Successful response - returns a single `Application`.
+      operationId: getApplication
+      summary: Get a Application
+      description: Gets the details of a single instance of a `Application`.
+    put:
+      requestBody:
+        description: Updated `Application` information.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Application'
+        required: true
+      responses:
+        "202":
+          description: Successful response.
+      operationId: updateApplication
+      summary: Update a Application
+      description: Updates an existing `Application`.
+    delete:
+      responses:
+        "204":
+          description: Successful response.
+      operationId: deleteApplication
+      summary: Delete a Application
+      description: Deletes an existing `Application`.
+    parameters:
+    - name: applicationId
+      description: A unique identifier for a `Application`.
+      schema:
+        type: string
+      in: path
+      required: true
+components:
+  schemas:
+    Application:
+      title: Root Type for Application
+      description: A Service Application
```

Comment:
Right now we have contract instead of application.

| Type | Description | Reference |
| ----- | ----------- | ---------- |
| Contract | Meta class for all types of contracts | Plan (Meta class) |
| Cinstance |  The actual application | ApplicationPlan |
| ServiceContract | A type of contract by Service | ServicePlan |
| AccountContract | A type of contract by Account | AccountPlan |